examples/gno.land/p/demo/boards/board.gno

```diff
@@ -0,0 +1,137 @@
+package boards
+
+import (
+	"std"
+	"strconv"
+	"time"
+
+	"gno.land/p/demo/avl"
+)
+
+//----------------------------------------
+// Board
+
+type BoardID uint64
+
+func BoardIDKey(bid BoardID) string {
+	return padZero(uint64(bid), 10)
+}
+
+func (bid BoardID) String() string {
+	return strconv.Itoa(int(bid))
+}
+
+type Board struct {
+	id        BoardID // only set for public boards.
+	url       string
+	name      string
+	creator   std.Address
+	threads   avl.Tree // Post.id -> *Post
+	postsCtr  uint64   // increments Post.id
+	createdAt time.Time
+	deleted   avl.Tree // TODO reserved for fast-delete.
+}
+
+/* TODO support this once we figure out how to ensure URL correctness.
+// A private board is not tracked by gBoards*,
+// but must be persisted by the caller's realm.
+// Private boards have 0 id and does not ping
+// back the remote board on reposts.
+func NewPrivateBoard(url string, name string, creator std.Address) *Board {
+	return newBoard(0, url, name, creator)
+}
+*/
+
+func NewBoard(id BoardID, url string, name string, creator std.Address) *Board {
+	return &Board{
+		id:        id,
+		url:       url,
+		name:      name,
+		creator:   creator,
+		threads:   avl.Tree{},
+		createdAt: time.Now(),
+		deleted:   avl.Tree{},
+	}
+}
+
+func (board *Board) Id() BoardID {
+	return board.id
+}
+
+func (board *Board) Url() string {
+	return board.url
+}
+
+func (board *Board) Name() string {
+	return board.name
+}
+
+func (board *Board) Creator() std.Address {
+	return board.creator
+}
+
+func (board *Board) Threads() avl.Tree {
+	return board.threads
+}
+
+func (board *Board) CreatedAt() time.Time {
+	return board.createdAt
+}
+
+func (board *Board) IsPrivate() bool {
+	return board.id == 0
+}
+
+func (board *Board) GetThread(pid PostID) *Post {
+	pidkey := postIDKey(pid)
+	postI, exists := board.threads.Get(pidkey)
+	if !exists {
+		return nil
+	}
+	return postI.(*Post)
+}
+
+func (board *Board) AddThread(creator std.Address, title string, body string) *Post {
```

Comment:
the name `thread` and `post` has been used interchangeably.

Reply:
`thread` is basically a `post` (feel free to correct me, I didn't get your concern)

I just copied it over without any modifications 🤷‍♂️

Reply:
yeah mostly nits,  I think you can keep it as it were.

Reply:
@ltzmaxwell could you approve this PR, or request changes if needed?